Snake/Snake/UILayer.cs

```diff
@@ -13,7 +13,7 @@ class Program
 
         static bool ExitGame = false;
         static string UserReplay;
-
+        enum Direction { right, down, left, up };

```

Comment:
You already have a Direction enum — I would strongly recommend not declaring it a second time. If you pull your Direction enum out of the Engine and have it in its own file in the BusinessLayer namespace, you should be able to refer to it here without explicitly declaring it?